client/src/components/Pagination.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect, useCallback } from "react";
+import { Pagination as PaginationBootstrap } from "react-bootstrap";
+import "./Pagination.css";
+
+const Pagination = ({
+  items,
+  totalPages,
+  paginationClass,
+  setActivePage,
+  isSearchActive,
+  activePage,
+  paginationReset,
+  setPaginationReset,
```

Comment:
when dealing with booleans, I suggest prefixing them with `is` or `should` to clearly show they are a boolean value

```suggestion
  shouldResetPagination,
  setShouldResetPagination,
```